app/models/taxon_facet.rb

```diff
@@ -118,7 +136,11 @@ def default_topic_value
     { text: "All topics", value: "" }
   end
 
-  def registry
-    @registry ||= Registries::BaseRegistries.new.all["part_of_taxonomy_tree"]
+  def full_topic_taxonomy
```

Comment:
Could we rename these (at the Registry level), so their usage is clearer? Maybe something for another PR, although we could use clearer names within this class. 

Example:

- `full_topic_taxonomy` -> `flat_topic_taxonomy_hash`
- `partial_topic_taxonomy` -> `upper_topic_taxonomy_tree`